src/rust/engine/options/src/lib.rs

```diff
@@ -335,18 +367,18 @@ impl OptionParser {
         let mut derivation = None;
         if self.include_derivation {
             let mut derivations = vec![(Source::Default, default.to_owned())];
-            for (source_type, source) in self.sources.iter().rev() {
+            for (source_type, source) in self.sources.iter() {
                 if let Some(val) = getter(source, id)? {
-                    derivations.push((*source_type, val));
+                    derivations.push((source_type.clone(), val));
```

Comment:
Right now the derivations will contain many hundreds of copies of the string `pants.toml` or whatever. In a followup I might change this to a `&str` with a lifetime tied to the OptionsParser, so we can re-use references.

Reply:
Would have to be Arc because these get passed into Python code, but yeah, not a bad solution.